### PR TITLE
Allows cyborgs and AI to see packet sniffer device output and set filters

### LINKED ISSUE #5159: Borgs and packet sniffers
Submitter: amazingdragons
**Describe the bug**
Borgs can't use packet sniffers

**To Reproduce**
1. Screwdriver a packet sniffer onto a data terminal
2. Click on it to use it
3. It doesn't work
**Expected behavior**
It should open up the packet sniffer info, but it doesn't

**Screenshots**

**Additional context**

Reported on: Goonstation 3 Roleplay: Morty 2021-06-15 15:58:51

### PR DESCRIPTION
**Pull request by 112358sam**
[bug-minor]

## About the PR 
This allows cyborgs and AI to view the output of the activated packet sniffer device and set the netid filtering.

## Why's this needed? 
Fixes #5159

## Changelog

```changelog
(u)112358sam
(+)Borgs and AI can now see packet sniffer device output and set filters.
```

### CODE REVIEW

File: code/modules/networks/computer3/sniffer.dm [hunk 1]
```diff
@@ -85,32 +87,32 @@
...
+			if (!t)
+				src.filter_id = null
+				src.updateIntDialog()
+				return
 
-				if(length(t) != 8 || !is_hex(t))
-					src.filter_id = null
-					src.updateIntDialog()
-					return
+			if (!issilicon(usr) && !isAIeye(usr) && (!in_interact_range(src, usr) || usr.stat || usr.restrained()))
```
Comment (ZeWaka) [inline code/modules/networks/computer3/sniffer.dm:106]: ambiguous grouping

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/networks/computer3/sniffer.dm (hunk 1 above)
Comment (112358sam) [reply code/modules/networks/computer3/sniffer.dm]: Fixed!

File: code/modules/networks/computer3/sniffer.dm [hunk 2]
```diff
@@ -85,32 +87,33 @@
 	Topic(href, href_list)
 		..()
 
-		if (usr.contents.Find(src) || usr.contents.Find(src.master) || (istype(src.loc, /turf) && get_dist(src, usr) <= 1))
+		if (!issilicon(usr) && !isAIeye(usr))
+			if (!usr.contents.Find(src) && !usr.contents.Find(src.master) && !(istype(src.loc, /turf) && get_dist(src, usr) <= 1))
```
Comment (ZeWaka) [inline code/modules/networks/computer3/sniffer.dm:91]:
```suggestion
			if (!(src in usr.contents) && !(src.master in usr.contents) && !(istype(src.loc, /turf) && IN_RANGE(src, usr, 1))
```

File: code/modules/networks/computer3/sniffer.dm [hunk 3]
```diff
@@ -85,32 +87,33 @@
 	Topic(href, href_list)
 		..()
 
-		if (usr.contents.Find(src) || usr.contents.Find(src.master) || (istype(src.loc, /turf) && get_dist(src, usr) <= 1))
+		if (!issilicon(usr) && !isAIeye(usr))
+			if (!(src in usr.contents) && !(src.master in usr.contents) && !(istype(src.loc, /turf) && IN_RANGE(src, usr, 1))
```
Comment (ZeWaka) [inline code/modules/networks/computer3/sniffer.dm:91]:
```suggestion
			if (!(src in usr.contents) && !(src.master in usr.contents) && !(istype(src.loc, /turf) && IN_RANGE(src, usr, 1)))
```
typo

**Pull request merged**